src/hydro_system.hpp

```diff
@@ -499,7 +526,9 @@ void HydroSystem<problem_t>::ComputeFlatteningCoefficients(amrex::MultiFab const
 		const double chi_min = std::max(0., std::min(1., (beta_max - beta) / (beta_max - beta_min)));
 
 		// Z is a measure of shock strength (Eq. 76 of Miller & Colella 2002)
-		const double K_S = gamma_ * P; // equal to \rho c_s^2
+		amrex::GpuArray<Real, nmscalars_> massScalars = RadSystem<problem_t>::ComputeMassScalars(primVar, i, j, k);
+		const double K_S = std::pow(quokka::EOS<problem_t>::ComputeSoundSpeed(primVar(i, j, k, primDensity_index), P, massScalars), 2) *
+				   primVar(i, j, k, primDensity_index);
```

Comment:
@BenWibking in this line, should the second argument to `ComputeSoundSpeed` be just `P` or `P*primVar(i,j,k,primDensity_index)`? We use the latter a few lines above in the same function.

Reply:
Hmm, okay. So, in this function, we first use `P` in line 489: `amrex::Real P = primVar(i, j, k, pressure_index);`, and at this time, `P` is specific Eint. Now, if `reconstruct_eint` is true, `P` is the pressure, but if its not true, it is still the specific Eint (for gamma > 1)? So by the time we come to this line (529), we don't know a priori if `P` is specific Eint or pressure?

Reply:
By the time we come to this line, `P` is always pressure.

Reply:
ah okay. so then what i have is correct.